rfcs/text/refactoring-date-component.md

```diff
@@ -0,0 +1,407 @@
+
+Another option is that we refactor the existing component to simplify the code and fix the issues already raised. However, we would still need to introduce breaking changes as we plan to refactor all class-based input components to be functional anyway. As well as this, refactoring could likely prove more complex than rewriting the component: some of the legacy code outdates any of the current Carbon team and is tightly coupled with older projects.
+
+In terms of alternate designs, we could look to alter the value emitted on event handling, other UI libraries emit a Date object for example. This was rejected as it would mean we would make migrating from the old implementation to the new component more difficult and may mean move away from what was defined in previous RFC documents. Currently our consuming projects expect to pass a string value and to receive one back, if we keep this pattern then we are not putting additional load on to them with respect to formatting the value from Date to string.
+
+There are several date management solutions available as an alternative to `date-fns`: [Luxon](https://moment.github.io/luxon/); [Dayjs](https://day.js.org/); [js-Joda](https://js-joda.github.io/js-joda/); or using the Date object directly instead of a library. An additional option would be to integrate an abstraction for any of these potential solutions, for example [date-io](https://github.com/dmtrKovalenko/date-io) would support this. Of these options, `date-fns` is the most lightweight and will require the least integration into the new component. It is either the chosen or default option for many other UI libraries. For our requirements, it is unlikely we will need to surface an abstraction at this point as we will not be requiring implementation teams to do any configuration past what is defined in the interface and what they will need to pass in via the locale. If the need to introduce an abstraction arose it would not be too complex to refactor it to support this without introducing a breaking change.
+
+Similarly there are a number of other options for integrating within the `DatePicker`, for example [react-datepicker](https://reactdatepicker.com/) and [react-flatpickr](https://github.com/haoxins/react-flatpickr). However, they all come with limitations: a lack of documentation, no support for CSS-in-JS and more configuration requirements with respect to locale support.
+
+# Adoption strategy
+
+As mentioned, this proposal will be implemented as a breaking change for the significant technical debt to be removed and to fix the existing bugs we have raised against the current implementation. In terms of how consuming projects will migrate to it, the majority of the interface should remain the same so the bulk of the work will be on adding the correct locale properties, so the desired formats are provided to the component.
```

Comment:
>  adding the correct locale properties, so the desired formats are provided to the component

This is no longer true.

Reply:
Again should have removed this 🤦

Reply:
Without the guard it throws just after mount as it's setting the `selectedValue` to undefined on the next render